ComponentKit/Core/Scope/CKComponentScopeFrame.mm

```diff
@@ -57,27 +57,31 @@ + (CKComponentScopeFramePair)childPairForPair:(const CKComponentScopeFramePair &
   CKCAssert([componentClass isSubclassOfClass:[CKComponent class]], @"%@ is not a component", NSStringFromClass(componentClass));
   CKAssertNotNil(pair.frame, @"Must have frame");
 
-  // Find the existing child, if any, in the old frame
-  CKComponentScopeFrame *existingChild;
+  CKComponentScopeFrame *existingChildFrameOfEquivalentPreviousFrame;
   if (pair.equivalentPreviousFrame) {
     const auto &equivalentPreviousFrameChildren = pair.equivalentPreviousFrame->_children;
     const auto it = equivalentPreviousFrameChildren.find({componentClass, identifier});
-    existingChild = (it == equivalentPreviousFrameChildren.end()) ? nil : it->second;
+    existingChildFrameOfEquivalentPreviousFrame = (it == equivalentPreviousFrameChildren.end()) ? nil : it->second;
   }
 
+  const auto existingChild = pair.frame->_children.find({componentClass, identifier});
+  CKComponentScopeHandleIdentifier existingGlobalIdentifier =
+  existingChild != pair.frame->_children.end() ? existingChild->second.handle.globalIdentifier : 0;
```

Comment:
Would anything break if you just did the following? 

It results in a pretty confusing situation where the map has two entries with the same `CKComponentScopeHandle`, but that actually makes sense I think. For example, we definitely want the state to be the same for each — not separate (created with `initialStateCreator` for the colliding component).

```
if (existingChild != pair.frame->children.end()) {
  return {.frame = existingChild->second, .equivalentPreviousFrame = existingChildFrameOfEquivalentPreviousFrame};
}
```